ablySpec/RealtimeClientChannel.swift

```diff
@@ -607,6 +607,55 @@ class RealtimeClientChannel: QuickSpec {
                     expect(Test.counter).toEventually(equal(2), timeout: testTimeout)
                 }
 
+                context("message cannot be decoded or decrypted") {
+
+                    // RTL7e
+                    it("should deliver with encoding attribute set indicating the residual encoding and error should be emitted") {
+                        let options = AblyTests.commonAppSetup()
+                        options.autoConnect = false
+                        let client = ARTRealtime(options: options)
+                        client.setTransportClass(TestProxyTransport.self)
+                        client.connect()
+                        defer { client.close() }
+
+                        let channelOptions = ARTChannelOptions(encrypted: ARTCrypto.defaultParams())
```

Comment:
This an incorrect definition for channel options for two reasons as far as I can see:

* A `key` value must always be provided as part of the `cipherParams`
* `encrypted` is a boolean, and `chiperParams`: should be the `cipherParams`

See http://docs.ably.io/client-lib-development-guide/features/#TB2 and second example at http://docs.ably.io/realtime/channels-messages/#channels